exporter/datadogexporter/internal/logs/sender_test.go

```diff
@@ -0,0 +1,220 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/testutil"
+)
+
+func TestSubmitLogs(t *testing.T) {
+	logger := zaptest.NewLogger(t)
+
+	tests := []struct {
+		name    string
+		payload []datadogV2.HTTPLogItem
+		testFn  func(jsonLogs testutil.JSONLogs, counter int)
+	}{
+		{
+			name: "same-tags",
+			payload: []datadogV2.HTTPLogItem{{
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 1",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 1",
+					"service":  "server",
+				},
+				AdditionalProperties: nil,
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 2",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 2",
+					"service":  "server",
+				},
+				AdditionalProperties: nil,
+			}},
+			testFn: func(jsonLogs testutil.JSONLogs, counter int) {
+				switch counter {
+				case 0:
+					assert.True(t, jsonLogs.HasDDTag("tag1:true"))
+					assert.Len(t, jsonLogs, 2)
+				default:
+					t.Fail()
+				}
+			},
+		},
+		{
+			name: "different-tags",
+			payload: []datadogV2.HTTPLogItem{{
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 1",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 1",
+					"service":  "server",
+				},
+				AdditionalProperties: nil,
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 2",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 2",
+					"service":  "server",
+				},
+				AdditionalProperties: nil,
```

Comment:
Please remove zero-value fields. This is already the default, no need to spell it out.